batcher/aligned-batcher/src/eth/mod.rs

```diff
@@ -72,3 +83,20 @@ pub async fn get_batcher_payment_service(
 
     Ok(service_manager)
 }
+
+pub fn signature_data_from_signature(signature: &Signature, nonce: [u8; 32]) -> SignatureData {
+    let mut r = [0u8; 32];
+    signature.r.to_big_endian(&mut r);
+
+    let mut s = [0u8; 32];
+    signature.s.to_big_endian(&mut s);
+
+    let nonce = U256::from_big_endian(nonce.as_slice());
+
+    SignatureData {
+        v: signature.v as u8,
+        r,
+        s,
+        nonce,
+    }
+}
```

Comment:
small nit, this can be a little more rusty with
```suggestion
impl SignatureData {
   pub fn from(value: &Signature, nonce: [u8; 32]) -> Self {
      ...
   }
}
```

Reply:
Changed on 9c847105c287437548423c40bf2d212b41ed02fa